### PR TITLE
PCHR-2322: Move Leave Request Popup 

### PR DESCRIPTION
**Pull request by deb1990**
## Overview
As part of this PR(and https://github.com/compucorp/org.civicrm.shoreditch/pull/20 and https://github.com/compucorp/civihr-employee-portal-theme/pull/179)The leave request popup has been injected in admin leave and absence and styles has been modified. So that when the grid cells of the calendar is clicked, the popup opens.

## Technical Details
1. Backstop JS tests are added.
2. `text-angular` and `leave-request-popup` are added in admin app for the leave popup and comment section to work.
3. In `leave-request-popup.js` 
  - `appendTo: $rootElement.children().eq(0)` added as a settings to the popup, as this is necessary for the popup to open in the admin context.
  - `windowClass: 'chr_leave-request-modal'` is added as a settings to the popup, which is necessary for styling.
4. Moved the scss from SSP theme
  -  `_leave-request-comment.scss`
  -  `_leave-request-files.scss`
  -  `_leave-request-modal.scss`
5. In `_absence-tab-page.scss`, `border-color` is overridden for the the `btn-group`, this is necessary to match the design for the following item
civihr_staff_compucorp_co_uk___hr17-1
6. In `leave.html, markup for the datepicker is changed according to the styleguide
```html
<span
	class="input-group-addon pointer"
	ng-disabled="$ctrl.isMode('view') || ($ctrl.isRole('manager') && !$ctrl.isMode('create') && !$ctrl.isLeaveType('sickness'))"
	ng-click="$ctrl.uiOptions.date.from.show = true">
	<i class="fa fa-calendar"></i>
</span>
```

## Comments
This PR should not be merged until https://github.com/compucorp/org.civicrm.shoreditch/pull/20 is merged and 1.7 is synced.

---

- [x] Tests Pass

### CODE REVIEW

File: uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-request-comment.scss [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.chr_leave-request-comment{
+  margin-top: 10px;
+}
+.chr_leave-request-comment__text {
```
Comment (bmnepali) [inline uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-request-comment.scss]: An empty line between selectors.

File: uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-request-files.scss [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  margin-top: 20px;
+  margin-bottom: 20px;
+}
+
+.chr_leave-request-files_meta {
+  margin-bottom: 10px;
+}
+
+.chr_leave-request-files_upload {
+  border: dashed 1px darkgray;
```
Comment (bmnepali) [inline uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-request-files.scss]: Do you mean `border: dashed 1px $gray-darker` here?

File: uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-request-modal.scss [hunk 3]
```diff
@@ -0,0 +1,104 @@
+$row-margin-bottom: 15px;
+
+.chr_leave-request-modal__chevron {
+  font-size: 0.6em;
+  vertical-align: middle;
+  margin-right: 5px;
```
Comment (bmnepali) [inline uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-request-modal.scss]: list  style attributes in alphabetical order for better readability.
```css
.chr_leave-request-modal__chevron {
    font-size: 0.6em;
    margin-right: 5px;
    vertical-align: middle;
}
```

File: uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-request-modal.scss [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+.chr_leave-request-modal__required-documents input {
+  margin-right: 5px;
+}
+
+.chr_leave-request-modal__small_spinner {
+
+  > div.chr_spinner {
+    min-height: initial;
+    margin-top: 18px;
```
Comment (bmnepali) [inline uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-request-modal.scss]: In alphabetical order.

File: uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-request-modal.scss [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  * It is hardcoded to 13px for now.
+  */
+  input[uib-datepicker-popup] {
+    font-size: 13px;
+  }
+}
+
+.chr_leave-request-modal__table {
+  margin-bottom: $row-margin-bottom !important;
+  border: 1px solid $crm-table-heading-border-color;
```
Comment (bmnepali) [inline uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-request-modal.scss]: In alphabetical order. Check throughout the file.

**Comment by bmnepali** [comment]
@deb1990 Also resolve the merge conflicts.

File: uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-request-files.scss (hunk 2 above)
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/scss/components/_leave-request-files.scss]: Well all these files are just moved from SSP, but I looked that changing it to `$gray-darker` is not making much difference in how it looks. So Done

**Pull request merged**